js/zb.js

```diff
@@ -2571,6 +2573,94 @@ module.exports = class zb extends Exchange {
         return result;
     }
 
+    async fetchLedger (code = undefined, since = undefined, limit = undefined, params = {}) {
+        if (code === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchLedger() requires a code argument');
+        }
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const request = {
+            'futuresAccountType': 1,
+            // 'currencyId': '11',
+            // 'type': 1,
+            // 'endTime': this.milliseconds (),
+            // 'pageNum': 1,
+        };
+        if (code !== undefined) {
+            request['currencyName'] = currency['id'];
+        }
+        if (since !== undefined) {
+            request['startTime'] = since;
+        }
+        if (limit !== undefined) {
+            request['pageSize'] = limit;
+        }
+        const response = await this.contractV2PrivateGetFundGetBill (this.extend (request, params));
+        //
+        //     {
+        //         "code": 10000,
+        //         "data": {
+        //             "list": [
+        //                 {
+        //                     "type": 3,
+        //                     "changeAmount": "0.00434664",
+        //                     "isIn": 0,
+        //                     "beforeAmount": "30.53353135",
+        //                     "beforeFreezeAmount": "21.547",
+        //                     "createTime": "1646121604997",
+        //                     "available": "30.52918471",
+        //                     "unit": "usdt",
+        //                     "symbol": "BTC_USDT"
+        //                 },
+        //             ],
+        //             "pageNum": 1,
+        //             "pageSize": 10
+        //         },
+        //         "desc": "操作成功"
+        //     }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const list = this.safeValue (data, 'list', []);
+        const result = [];
+        for (let i = 0; i < list.length; i++) {
+            const entry = list[i];
+            const timestamp = this.safeString (entry, 'createTime');
+            let direction = undefined;
+            const changeDirection = this.safeNumber (entry, 'isIn');
+            if (changeDirection === 1) {
+                direction = 'increase';
+            } else {
+                direction = 'reduce';
+            }
+            let fee = undefined;
+            const feeCost = this.safeNumber (entry, 'fee');
+            if (feeCost !== undefined) {
+                fee = {
+                    'cost': feeCost,
+                    'currency': this.safeCurrencyCode (this.safeString (entry, 'unit')),
+                };
+            }
+            result.push ({
```

Comment:
Let's move the parsing stage to `parseLedgerEntry` (see how it's done in the other exchanges).